### DISCUSSION TITLE
meaning of token probebility

### DISCUSSION THREAD

**Discussion by ofersa** [ggml-org/whisper.cpp#289] [Q&A]
Hi,

I try to understand the probability mechanism of a token.
in the python whisper project, the probability is an average of the whole segment and it should be less then -1.0 in order to be good enough.
in the whisper.cpp project the probability is per token that is usually one word and it range between 0.0 and 1.0 in order to be good enough. the values of whisper.cpp looks much more compliant with the audio.
how is it calculated, and why are the differ ?

Thanks
Ofer

**Comment by ggerganov** [comment]
Currently, the token probability is simply the soft-max of the logits produced by the decoder:

https://github.com/ggerganov/whisper.cpp/blob/1d716d6e34f3f4ba57bd9706a9258a0bdb008153/whisper.cpp#L1825-L1844
